### ISSUE TITLE
Config.alias_generator is ignored in __init__ parameter

### ISSUE THREAD

**Issue by skewty** [koxudaxi/pydantic-pycharm-plugin#78]
```py
class FooBase(BaseModel):
    x: str
    y: str = Schema(..., alias='special')

    class Config:
        alias_generator = str.upper

class SuperFoo(FooBase):
    z: str

SuperFoo(X='field name is now upper case', special='Do not forget about alias in schema', Z='Zebra')
```

**Comment by koxudaxi** [comment]
Thank you for creating an issue.
I thought about this issue.
It may be challenging to implement. 
`alias_generator` has to evaluate field name on python interpreter.
However, I don't know how to call python function by kotlin code.
I will search for the way.

**Comment by dmontagu** [comment]
This feels to me like it might cause more pain than it’s worth. I definitely would want pycharm to always show me the field name for auto completion, not the alias. But I always leave allow population by alias on.

**Comment by koxudaxi** [comment]
@dmontagu 
`allow_population_by_field_name` is off  in default.
I think It's right. if `allow_population_by_field_name` is off then, use an alias filename.
What is a problem in this case?
I don't mean I can't agree. I want to know your thinking about it.

**Comment by dmontagu** [comment]
If you can detect it properly I think it’s fine, I’m just worried about handling edge cases if you try to actually evaluate the alias generator during static analysis.

**Comment by koxudaxi** [comment]
I have understood you are worried about it.
If I implement the feature then, I will do it carefully.
Thank you.

**Comment by skewty** [comment]
@dmontagu is this a pydantic bug?  Perhaps I should close this bug.

```py
from pydantic import BaseModel, Schema  # version 0.32.2

class Foo(BaseModel):
    name: str = Schema(..., alias="NAME")

Foo(name="Scott")  # unexpected argument
```
The docs state:

> allow_population_by_alias
    whether or not an aliased field may be populated by its name as given by the model attribute, rather than strictly the alias; please be sure to read the warning below before enabling this (default: False)

So `allow_population_by_alias` is `False` in the example above, but population by alias seems required.

**Comment by dmontagu** [comment]
Yeah, this is pretty unfortunate:

```python
from pydantic import BaseModel, Schema

class Foo(BaseModel):
    name: str = Schema(..., alias="NAME")

    class Config:
        allow_population_by_alias = True

Foo(name="Scott")  # unexpected argument
```

It would be nice if there was a plugin-configuration setting to "always use field names instead of aliases" or similar. That might be a nice workaround that wouldn't require perfectly solving the logic for inferring the value of `allow_population_by_alias`.

(I think there are probably a lot of people like me who want to use pythonic naming in their init calls to their models, while still supporting loads/dumps in camelcase for json support, so it doesn't seem crazy to have the option to override the use of alias in the init everywhere.)

(Also, on a related note, the old warning about enabling `allow_population_by_alias` has been removed from the docs.)

**Comment by koxudaxi** [comment]
I think consistency is essential. Also, it needs to able to control behavior by the user. 
OK, I will create a setting panel on the plugin first.
I will spend time to learn to implement a setting panel. Please wait for a while.

**Comment by skewty** [comment]
> It would be nice if there was a plugin-configuration setting to "always use field names instead of aliases" or similar. That might be a nice workaround that wouldn't require perfectly solving the logic for inferring the value of `allow_population_by_alias`.
> 
> (I think there are probably a lot of people like me who want to use pythonic naming in their init calls to their models, while still supporting loads/dumps in camelcase for json support, so it doesn't seem crazy to have the option to override the use of alias in the init everywhere.)

I agree.  I am on the verge of turning this **pydantic** add-on off and it is likely others will get annoyed as well.  I dislike the usage of `# noinspection PyArgumentList` all over in the code.  It can so easily hide real code issues after refactoring.

I think a primitive solution might suffice for most people without adding too much code complexity to the add-on.  Can adding a special comment to a .py file be caught by an add-on and enable the behavior you suggested above?  I don't really like the idea of a *global* setting.

Re-opening to invite further discussion / possible solution.

**Comment by koxudaxi** [comment]
> Can adding a special comment to a .py file be caught by an add-on and enable the behavior you suggested above?

I have never parsed comments. but, I will be able to get comment from a line. (eg `# type: ignore`)
However, I think special comment is ununderstandable easy for users.
If there is a readable special comment then, It's a good idea.

**Comment by dmontagu** [comment]
(Below, I'm going to refer to the `allow_population_by_alias` setting as `allow_population_by_field_name`, the new name of the setting in v1.0, since it actually reflects the semantics and so is less confusing.)

Here's a proposal:
1. If `allow_population_by_field_name` is `True`, always use the field names in the generated signatures (I can justify this choice if necessary, but won't bother if we're all on the same page)
2. If `allow_population_by_field_name` is `False` and `alias_generator` is explicitly set, use the inferred field names (which will be aliases if specified in the `Field`, the way it works now) in the generated signature, but *also* add a `**kwargs` argument to the generated signature to prevent pycharm warnings due to the use of generated aliases if they can't easily be inferred.
    1. Note: the `**kwargs` argument should *only* be added if `allow_population_by_field_name` is `False`! Otherwise, you should use the field names.

It seems to me that this 1) shouldn't be a great deal of work to implement, and 2) addresses most use cases very well (including both @skewty's and my own)

-----

@skewty if you take any issues with the  above, could you describe the specific behavior you would like from the plugin? By that, I mean examples of:

1. code that *should* raise warnings
1. code that *should not* raise warnings
1. how the configuration options you have in mind would change the displayed warnings

I think that would make it easier to discuss concretely.

**Comment by skewty** [comment]
@dmontagu I would be okay with such a compromise and would welcome a release with that implemented :)

PS: thank you all for all the time and energy that has been put into this plugin.

**Comment by koxudaxi** [comment]
@skewty @dmontagu 
Sorry, I was busy for a few days.
I want to tell you two things.

1. I have found to run the python interpreter by Kotlin.
However, the method creates a virtual file of python source code and calls the interpreter with the file.
I think It spends a lot of CPU time.
If we call python's interpreter then, we should create the python server to evaluate python code on the fly.
But, the idea needs time to develop.

2. I have created PR which supports allow_population_by_field_name 
https://github.com/koxudaxi/pydantic-pycharm-plugin/pull/82/files

I think right that We will merge about `2.` into master. Also, I would like to implement `1.` after next time.

Would you please give me your thoughts?

Btw, I will implement another feature which is setting panel or autocompletion of `Config` next.
